### ISSUE TITLE
bug in uploadedfileinput widget

### ISSUE THREAD

**Issue by ozgur-as** [jrief/django-formset#96]
hi there,

in my own project, i am using a collection to be able to add multiple images in a single form. i have come across a bug, so i wanted to reproduce it in the testapp with a simple gallery example. you can access the branch from: https://github.com/ozgur-as/django-formset/tree/reproduce-file-input-widget-bug

the example works as it should if you are just uploading for the first time/deleting/updating and doing a form submit after each action, but if you do the following actions without submitting the form the bug occurs:

- "add image" <- first image
- "choose file, browse, do the temporary upload" <- first image
- "add image" <- second image
- "choose file, browse, do the temporary upload" <- second image
- "delete the first image"
- "click add image"

browser console throws the following javascript error:
```
django-formset.js:25 Uncaught Error: More than one form has id="id_images.2.image"
    at dr.assignFieldsToForms (django-formset.js:25:27803)
    at HTMLButtonElement.appendFormCollectionSibling (django-formset.js:25:24025)
```
the widget's progress bar and choose file button stays unresponsive and submitted form would not save properly because of the mixed id's i suppose.

i hope it's a clear description of the steps :)

thanks in advance!

**Comment by jrief** [comment]
so which view/URL are you using then?

**Comment by ozgur-as** [comment]
i was using /default/gallerycollection

**Comment by jrief** [comment]
I can confirm this bug and believe to have found a solution.

Please checkout my branch at
https://github.com/jrief/django-formset/tree/reproduce-file-input-widget-bug
remember to recompile.

Thanks for reporting and please tell me if this works.

**Comment by ozgur-as** [comment]
yup, it looks fine to me :) thanks!

**Comment by jrief** [comment]
Released in version 1.3.3
